src/node/tests/elder.rs

```diff
@@ -214,19 +214,56 @@ impl Env {
         )
     }
 
+    fn self_cast_unordered_votes(&mut self, votes: impl IntoIterator<Item = Vote>) -> Result<()> {
+        let index = self.elders_info.position(self.subject.name()).unwrap();
+        let addr = ([127, 0, 0, 1], 9000 + index as u16).into();
+        let pk = self.public_key_set.public_key();
+        let secret_key_share = self.subject.secret_key_share()?.clone();
+        let proof_chain = self.subject.our_history().cloned();
+        let full_id = self.subject.full_id().clone();
+
+        for vote in votes {
+            info!(
+                "Vote as {:?} for {:?} ({:?})",
+                full_id.public_id(),
+                vote,
+                pk
+            );
+
+            let proof_share = vote.prove(self.public_key_set.clone(), index, &secret_key_share)?;
+            let variant = Variant::Vote {
+                content: vote.clone(),
+                proof_share,
+            };
+            let message = Message::single_src(
+                &full_id,
+                DstLocation::Section(*full_id.public_id().name()),
+                variant,
+                proof_chain.clone(),
+                Some(pk),
+            )?;
+
+            test_utils::handle_message(&mut self.subject, addr, message)?;
+        }
+
+        Ok(())
+    }
+
     fn accumulate_our_key_and_section_info_if_vote(
         &mut self,
         new_info: &DkgToSectionInfo,
     ) -> Result<()> {
+        let our_key_vote = Vote::OurKey {
+            prefix: new_info.new_elders_info.prefix,
+            key: new_info.new_pk_set.public_key(),
+        };
+        let section_info_vote = Vote::SectionInfo(new_info.new_elders_info.clone());
+
+        self.self_cast_unordered_votes(vec![our_key_vote.clone(), section_info_vote.clone()])?;
```

Comment:
This is still wrong. This function is not supposed to vote for the `subject` node. The idea is that something else has already triggered that vote and here we are only voting by the other nodes so together those votes accumulate. So if the tests fail without this it means that `subject` did not cast the vote it was supposed to. My guess is that the `simulate_dkg` function is probably not working as expected. Probably because it does not actually simulate receiving enough `DKGResult` messages from the other nodes.

Reply:
Hmm...  right. 
How about I just remove this fixing commit and merge PR 2188 only with the other two.
And trying to resolve this in a separate PR ?